mypy/checkexpr.py

```diff
@@ -351,7 +351,9 @@ def check_call(self, callee: Type, args: List[Expression],
         """
         arg_messages = arg_messages or self.msg
         if isinstance(callee, CallableType):
-            if callee.is_concrete_type_obj() and callee.type_object().is_abstract:
+            if (callee.is_type_obj() and callee.type_object().is_abstract
+                    # Exceptions for Type[...] and classmethod first argument
+                    and not callee.from_type_type and not callee.is_classmethod_class):
```

Comment:
Maybe also except static methods?

I found a case in our codebase that goes basically like this:
```python
class Logger:
    @staticmethod
    def log(a: Type[C]): ...
class C:
    @classmethod
    def action(cls) -> None:
        Logger.log(cls)  #E: Only non-abstract class can be given where 'Type[C]' is expected
    @abstractmethod
    ...
```
(I'm far from positive that it's because of this line though.)